data_tools/firebase_client_wrapper.py

```diff
@@ -303,10 +303,18 @@ def delete_dataset(dataset_id):
 def _delete_collection(coll_ref, batch_size):
```

Comment:
I'm not convinced that this method of deleting collections is either safe or cost-effective, because of Firebase's orphaned collection problem. This was the reason why we haven't actually been using it, but have used the console instead. 

Basically - if the docs in the collection contain subcollections, I think this method will orphan them. 

We could either try and understand what the recommended approach to doing this is, or we should delete this function from the fcw

Reply:
Sadly there is no easy way to recursively delete a document. Firebase's recommendation appears to be to do it in a cloud function (which has access to a recursive delete API). Another option is to reimplement the CLI's delete function in python.

This does work correctly at the moment when operating on valid datasets. It would only break if we uploaded an invalid dataset or if we updated the rest of data tools rather than just here. However, backup and restore suffer the same problems...